### PR TITLE
Add MaaS checks for processes status (#1366)

### PR DESCRIPTION
**Pull request by cloader89**
This adds a MaaS check script to check by fully qualified or basename
if processes are running on a host or within a container on a host.
Additionally, MaaS checks for beaver, elasticsearch and rsyslog are
added using the process check script.

Connects #1236
(cherry picked from commit 2870f4e)

### CODE REVIEW

**Comment by cloader89** [comment]
recheck_all

**Comment by alextricity25** [comment]
Hi, @cloader89.
Given that this is a feature enhancement PR, I don't think it's needed to backport this to Liberty.
If others feel it is necessary, then some changes will need to be made to ensure the plugins are not using the virtual environment wrapper script. MaaS plugins don't run in virutal environments in Liberty.

**Comment by cloader89** [comment]
recheck_all

**Comment by cloader89** [comment]
recheck_all

**Comment by git-harry** [comment]
👎 The commit message does not reference the original commmit.

To cherry-pick a commit into your current branch you should do:

```
git cherry-pick -x commit
```

Adding '-x' appends a line to the commit message referencing the original commit.

**Comment by cloader89** [comment]
recheck_all

**Comment by cloader89** [comment]
recheck_all

**Comment by cloader89** [comment]
recheck_upgrade

**Comment by cloader89** [comment]
recheck_all

**Comment by cloader89** [comment]
recheck_all

**Comment by cloader89** [comment]
18:20:21 TASK: [Verify Check & Alarm Status] ******************************************* 
18:50:21 Build timed out (after 30 minutes). Marking the build as aborted.
18:50:23 Build was aborted
recheck_all

**Comment by cloader89** [comment]
recheck_all

**Comment by cloader89** [comment]
recheck_all

File: releasenotes/notes/add-maas-checks-for-filbeat-es-rsyslog-0289a1d9a1c198d0.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    This adds MaaS checks for elasticsearch, beaver, and
```
Comment (sigmavirus24) [inline releasenotes/notes/add-maas-checks-for-filbeat-es-rsyslog-0289a1d9a1c198d0.yaml]: Thank you for updating the release note but it's my understanding that liberty and earlier don't use release notes. Is that correct @git-harry ?

**Comment by sigmavirus24** [comment]
One nit, but otherwise this looks good. I'll spin up a liberty AIO today and test it out.

File: releasenotes/notes/add-maas-checks-for-filbeat-es-rsyslog-0289a1d9a1c198d0.yaml (hunk 1 above)
Comment (git-harry) [inline releasenotes/notes/add-maas-checks-for-filbeat-es-rsyslog-0289a1d9a1c198d0.yaml]: Yes, tracking release notes within the repo was introduced with Mitaka. A few have already slipped through into previous branches but they will not be used in any way.

Comment (cloader89) [inline releasenotes/notes/add-maas-checks-for-filbeat-es-rsyslog-0289a1d9a1c198d0.yaml]: ok, will delete the release notes on this and the kilo PR as well.

**Comment by cloader89** [comment]
removed release notes @sigmavirus24 @git-harry for both liberty/kilo

**Comment by cloader89** [comment]
tempest failure recheck_upgrade

**Comment by d34dh0r53** [comment]
Will need to be squashed, other than that 👍

**Comment by cloader89** [comment]
recheck_all

**Comment by cloader89** [comment]
recheck_ceph

**Comment by Frank-ZhangXin** [comment]
LGTM 👍

**Comment by sigmavirus24** [comment]
👍

**Pull request merged**